packages/communication-react/review/beta/communication-react.api.md

```diff
@@ -2515,21 +2515,32 @@ export interface FileDownloadError {
 export type FileDownloadHandler = (userId: string, fileMetadata: FileMetadata) => Promise<URL | FileDownloadError>;
 
 // @beta
-export interface FileMetadata {
+export type FileMetadata = FileSharingMetadata | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ FileSharingMetadataWithOptions | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ ImageFileMetadata;
+
+// @beta (undocumented)
+export type FileMetadataAttachmentType = 'fileSharing' | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ 'fileSharingWithOptions' | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ 'inlineImage' | /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ 'attachedImage' | 'unknown';
+
+// @beta
+export interface FileSharingMetadata {
+    attachmentType: 'fileSharing';
+    extension: string;
+    name: string;
+    url: string;
+}
+
+// @beta
+export interface FileSharingMetadataWithOptions {
```

Comment:
FileSharingMetadataWithOptions is an extended interface of FileSharingMetadata, but only with one more id and options

Is it possible to make id optional? then directly add "options" into FileSharingMetadata so we don't need FileSharingMetadataWithOptions?

BTW, the property name options sounds a bit not accurate, it's more like metadata or payload